api/common.go

```diff
@@ -14,6 +14,54 @@ import (
 	"github.com/docker/libtrust"
 )
 
+type Command struct {
+	Name        string
+	Description string
+}
+
+var DockerCommands = map[string]Command{
```

Comment:
Would it make sense to keep DockerCommands defined as you have it, but instead of populating it this way where "attach" is listed twice (as the key of the map and as Command.Name), just have a temporary structure that has just a slice of Commands and we loop over that slice doing: `DockerCommands = append(DockerCommands, cmd[i].Name, cmd)`.   Its no big deal if you don't want to, I just get nervous when we I see duplicate data and expect people to keep them in-sync, or expect fat fingers to never introduce a typo.  :-)